File: src/tools/sites_snapshot/siteUtils.ts
import {log} from "../../utils/mcpLogger.js";
import {LoggingLevelSchema} from "@modelcontextprotocol/sdk/types.js";

export function isValidResponse(accountId: string, response: any): boolean {

    if (response.data?.accountSnapshot?.sites) {
        return true;
    }

    log(LoggingLevelSchema.Enum.debug, `No sites found in account snapshot for account ID: ${accountId}`);
    return false;
}

export function emptySitesResponse(timestamp: string): any {
    return {
        "data": {
            "accountSnapshotTimestamp": timestamp,
            "sites": []
        },
    }
}

File: python/src/tools/sites_snapshot/site_utils.py
"""
Site utilities
"""
from typing import Any, Dict
from ...utils.mcp_logger import log
from mcp import LoggingLevel  # type: ignore[import-untyped]


def is_valid_response(account_id: str, response: Dict[str, Any]) -> bool:
    """
    Validate response for site snapshot queries.
    
    Args:
        account_id: Account ID
        response: Response dictionary
        
    Returns:
        True if valid, False otherwise
    """
    if response.get("data", {}).get("accountSnapshot", {}).get("sites"):
        return True
    
    log("debug", f"No sites found in account snapshot for account ID: {account_id}")
    return False


def empty_sites_response(timestamp: str) -> Dict[str, Any]:
    """
    Create empty sites response.
    
    Args:
        timestamp: Timestamp string
        
    Returns:
        Empty sites response
    """
    return {
        "data": {
            "accountSnapshotTimestamp": timestamp,
            "sites": []
        },
    }

